src/lib/y2users/config_mergers/base.rb

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "abstract_method"
+
+module Y2Users
+  module ConfigMergers
+    # Helper class to merge users and groups of two configs
+    class Base
+      # Constructor
+      #
+      # @param lhs [Config] Left Hand Size config
+      # @param rhs [Config] Right Hand Size config
+      def initialize(lhs, rhs)
+        @lhs = lhs
+        @rhs = rhs
+      end
+
+      def merge
+        elements = rhs.users + rhs.groups
+
+        elements.each { |e| merge_element(lhs, e) }
+      end
+
+    private
+
+      # @return [Config] Left Hand Size config
+      attr_reader :lhs
+
+      # @return [Config] Right Hand Size config
+      attr_reader :rhs
+
+      abstract_method :merge_element
```

Comment:
to be honest I do not like this. E.g. Autoyast merger needs specific strategy for users and groups, so then I would need in abstract method to again separate them. Why not simply make merge method abstract?

Reply:
Well, I think that `#merge` itself is always the same: to iterate all elements and to merge each element into `lhs`. So, for me, it is something shared by all the mergers. Only the logic to merge an element into a config is what changes among the mergers. On other hand, if autoyast requires to distinguish the type of element, I guess it can do it just inside its `#merge_element` method.